x-pack/plugins/stack_connectors/common/microsoft_defender_endpoint/schema.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION } from './constants';
+
+// ----------------------------------
+// Connector setup schemas
+// ----------------------------------
+export const MicrosoftDefenderEndpointConfigSchema = schema.object({
+  clientId: schema.string({ minLength: 1 }),
+  tenantId: schema.string({ minLength: 1 }),
+  oAuthServerUrl: schema.string({ minLength: 1 }),
+  oAuthScope: schema.string({ minLength: 1 }),
+  apiUrl: schema.string({ minLength: 1 }),
+});
+export const MicrosoftDefenderEndpointSecretsSchema = schema.object({
+  clientSecret: schema.string({ minLength: 1 }),
+});
+
+// ----------------------------------
+// Connector Methods
+// ----------------------------------
+export const MicrosoftDefenderEndpointDoNotValidateResponseSchema = schema.any();
+
+export const MicrosoftDefenderEndpointBaseApiResponseSchema = schema.maybe(
+  schema.object({}, { unknowns: 'allow' })
+);
+
+export const TestConnectorParamsSchema = schema.object({});
+
+export const AgentDetailsParamsSchema = schema.object({
+  id: schema.string({ minLength: 1 }),
+});
+
+export const IsolateHostParamsSchema = schema.object({
+  id: schema.string({ minLength: 1 }),
+  comment: schema.string({ minLength: 1 }),
```

Comment:
Not sure we need it. We're mirroring the `isolate` action to be the same as the `isolate` for endpoint. I don't know how `Selective` works and my guess is it would change how we expose `isolate` for MS Defender. 

For now, we'll go with `Full` only and if we nee to support `Selective`, then we'll need to treat that as an addon feature later.

Reply:
That sounds good to me.